src/main/kotlin/app/hashers/CommitHasher.kt

```diff
@@ -73,18 +102,33 @@ class CommitHasher(private val localRepo: LocalRepo,
                     total + file.getAllAdded().size }
                 new.numLinesDeleted = diffFiles.fold(0) { total, file ->
                     total + file.getAllDeleted().size }
-
                 new
             }
             .observeOn(Schedulers.io())  // Different thread for data sending.
             .buffer(20, TimeUnit.SECONDS)  // Group ready commits by time.
-            .doOnNext { commitsBundle ->  // Send ready commits.
-                postCommitsToServer(commitsBundle) }
-            .blockingSubscribe({
-                // OnNext
-            }, { t ->  // OnError
-                Logger.error("Error while hashing: $t")
-                t.printStackTrace()
+            .blockingSubscribe({ commitsBundle ->  // Send ready commits.
+                postCommitsToServer(commitsBundle)
+            }, { e ->
+                Logger.error("Error while hashing: $e")
+            }, {
+                val facts = mutableListOf<Fact>()
+                factsDayTime.map { (author, list) ->
+                    list.forEachIndexed { hour, count ->
+                        if (count > 0) {
+                            facts.add(Fact(repo, FACT_KEY_COMMITS_DAYTIME +
+                                hour, count.toDouble(), author))
+                        }
```

Comment:
curious, if FACT_KEY_BLABLA_NUMBER can be sent to the server in a form to avoid strings parsing (regexps?), for example, if there's a way to serialize a list so that the server may deserialize it by some standard API.

Reply:
Me too.